test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -88,4 +92,50 @@ static public void testPassingNullToIsFlattenedThrowsException() throws Throwabl
 			myUnsafe.isFlattened(null);
 		});
 	}
+
+	@Test
+	static public void testValueHeaderSizeOfValueType() throws Throwable {
+		List<String> arguments = ManagementFactory.getRuntimeMXBean().getInputArguments();
+		if (arguments.contains("-Xnocompressedrefs")) {
+			assertEquals(myUnsafe.valueHeaderSize(ValueTypePoint2D.class), 8);
+		} else {
+			assertEquals(myUnsafe.valueHeaderSize(ValueTypePoint2D.class), 4);
+		}
+	}
+
+	@Test
+	static public void testValueHeaderSizeOfPaddedValueType() throws Throwable {
```

Comment:
Please add a comment saying that the padding size is included in the header size so it is checking for 8.

Reply:
addressed in 780435c

Reply:
It is not always padded, so it is better to rename t`estValueHeaderSizeOfPaddedValueType` to `testValueHeaderSizeOfClassWithLongField`